aea/aea_builder.py

```diff
@@ -1211,78 +1225,10 @@ def _load_and_add_components(
                 component = self._component_instances[component_type][configuration]
             else:
                 configuration = deepcopy(configuration)
-                component = load_component_from_config(component_type, configuration)
-            resources.add_component(component)
-
-    def _load_and_add_skills(self, context: AgentContext, resources: Resources) -> None:
-        for configuration in self._package_dependency_manager.skills.values():
-            logger_name = "aea.packages.{}.skills.{}".format(
-                configuration.author, configuration.name
-            )
-            configuration = cast(SkillConfig, configuration)
-            if configuration in self._component_instances[ComponentType.SKILL].keys():
-                skill = cast(
-                    Skill, self._component_instances[ComponentType.SKILL][configuration]
-                )
-                skill.skill_context.set_agent_context(context)
-                skill.skill_context.logger = logging.getLogger(logger_name)
-            else:
-                configuration = deepcopy(configuration)
-                skill_context = SkillContext()
-                skill_context.set_agent_context(context)
-                skill_context.logger = logging.getLogger(logger_name)
-                skill = cast(
-                    Skill,
-                    load_component_from_config(
-                        ComponentType.SKILL, configuration, skill_context=skill_context
-                    ),
+                component = load_component_from_config(
+                    component_type, configuration, **kwargs
                 )
-            resources.add_component(skill)
-
-    def _load_connection(
-        self, identity: Identity, wallet: Wallet, configuration: ConnectionConfig
-    ) -> Connection:
-        """
-        Load a connection from a directory.
-
-        :param identity: the AEA identity
-        :param wallet: the wallet
-        :param configuration: the connection configuration.
-        :return: the connection.
-        """
-        if configuration in self._component_instances[ComponentType.CONNECTION].keys():
-            connection = cast(
-                Connection,
-                self._component_instances[ComponentType.CONNECTION][configuration],
-            )
-            if connection.address != identity.address:
-                logger.warning(
-                    "The address set on connection '{}' does not match the default address!".format(
-                        str(connection.connection_id)
-                    )
-                )
```

Comment:
notice, we are not checking this anywhere anymore.

Reply:
I don't think we have to as the loading mechanism ensures the address comes from the identity (i.e. we pass the identity into the connection!)